CMakeLists.txt

```diff
@@ -141,7 +144,7 @@ function(preprocess_def_file inputFilename outputFilename)
 
   add_custom_command(
     OUTPUT ${outputFilename}
-    COMMAND ${CMAKE_CXX_COMPILER} /P /EP /TC ${PREPROCESS_DEFINITIONS}  /Fi${outputFilename}  ${inputFilename}
+    COMMAND cl.exe /P /EP /TC ${PREPROCESS_DEFINITIONS}  /Fi${outputFilename}  ${inputFilename}
```

Comment:
We should not hardcode the cl.exe. What is the problem with using the CMAKE_CXX_COMPILER?